src/index.jsx

```diff
@@ -5,78 +5,178 @@ import * as skinview3d from 'skinview3d';
+            // animation
+            if (run || walk) {
+                viewer.animation = new skinview3d.CompositeAnimation();
+                if (run) {
+                    this.createAnimation(viewer.animation, true);
+                }
+                if (walk) {
+                    this.createAnimation(viewer.animation, false);
+                }
+            }
+
+            // let's call ready here
+            this.props.onReady(this.state.viewer);
+        });
+    }
+
+    componentWillUnmount() {
+        this.setState({
+            viewer: null,
+            walk: null,
+            run: null,
+        });
     }
-  }
 
-  render() {
-    return (
-      <div className={this.props.className} ref={this.skinviewRef} style={{ imageRendering: 'pixelated' }}></div>
-    )
-  }
+    componentDidUpdate(prevProps) {
+        const { viewer, run, walk } = this.state;
+
+        if (prevProps.skinUrl !== this.props.skinUrl) {
+            viewer.skinUrl = this.props.skinUrl;
+        }
+
+        if (prevProps.capeUrl !== this.props.capeUrl) {
+            viewer.capeUrl = this.props.capeUrl;
+        }
+
+        if (prevProps.width !== this.props.width || prevProps.height !== this.props.height) {
+            viewer.setSize(this.props.width, this.props.height);
+        }
+
+        // animation
+        if (prevProps.run !== this.props.run) {
+            if (this.props.run) {
+                viewer.animation = new skinview3d.CompositeAnimation();
+                return this.createAnimation(viewer.animation, true);
+            } else {
+                if (run) run.remove();
+            }
+        }
+        if (prevProps.walk !== this.props.walk) {
+            if (this.props.walk) {
+                viewer.animation = new skinview3d.CompositeAnimation();
+                return this.createAnimation(viewer.animation, false);
+            } else {
+                if (walk) walk.remove();
+            }
+        }
+
+        if (prevProps.runSpeed !== this.props.runSpeed) {
+            if (run) {
+                run.speed = this.props.runSpeed;
+                this.setState({run});
+            }
+        }
+        if (prevProps.walkSpeed !== this.props.walkSpeed) {
+            if (walk) {
+                walk.speed = this.props.walkSpeed;
+                this.setState({walk});
+            }
+        }
+
+        if (prevProps.runPaused !== this.props.runPaused) {
+            if (run) {
+                run.paused = this.props.runPaused;
+                this.setState({run});
+            }
+        }
+        if (prevProps.walkPaused !== this.props.walkPaused) {
+            if (walk) {
+                walk.paused = this.props.walkPaused;
+                this.setState({walk});
+            }
+        }
+
+        if (prevProps.paused !== this.props.paused) {
+            viewer.animationPaused = this.props.paused;
+            this.setState({viewer});
+        }
+    }
+
+    createAnimation(viewerAnimation, isRun) {
```

Comment:
My only other piece of feedback would be to avoid adding the creation of the animation in this lib. A developer could just implement this via the `onReady` callback.

I was actually working on this recently, let me push my changes.

Edit: https://github.com/Hacksore/react-skinview3d/blob/master/src/stories/skinview3d.stories.jsx#L79

Reply:
OK, what do I need to do, close the PR?

Reply:
I like a lot of the other things you did. So maybe just rip out the animation and I will approve.